### ISSUE TITLE
Unable to build AppImageKit on armhf

### ISSUE THREAD

**Issue by AIIX** [AppImage/AppImageKit#765]
[ 58%] Building C object src/CMakeFiles/appimagetool.dir/binreloc.c.o
[ 59%] Building C object src/CMakeFiles/appimagetool.dir/elf.c.o
[ 60%] Linking C executable appimagetool
../squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/.libs/libsquashfuse.a(libsquashfuse_la-decompress.o): In function `sqfs_decompressor_zlib':
/home/phablet/AppImageKit/build/squashfuse-EXTERNAL-prefix/src/squashfuse-EXTERNAL/decompress.c:42: undefined reference to `uncompress'
collect2: error: ld returned 1 exit status
src/CMakeFiles/appimagetool.dir/build.make:144: recipe for target 'src/appimagetool' failed
make[2]: *** [src/appimagetool] Error 1
CMakeFiles/Makefile2:1221: recipe for target 'src/CMakeFiles/appimagetool.dir/all' failed
make[1]: *** [src/CMakeFiles/appimagetool.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2

Followed:
- git clone: https://github.com/AppImage/AppImageKit
- ./install-build-deps.sh 
- ./build.sh

System Info:
- Kernel: Linux localhost 3.10.73-lineageos-g0b64b0cd08b-dirty #3 SMP PREEMPT Thu Oct 5 19:35:59 IST 2017 aarch64 aarch64 aarch64 GNU/Linux
- Distro: Ubuntu Xenial 16.04
- Architecture: armhf

**Comment by TheAssassin** [comment]
What _exact_ commit are you trying to build at the moment? This linker error has occurred recently due to a bug that should be fixed now.

Also, please verify `zlib1g-dev` is installed. It might be missing in the `install-deps.sh` file.

**Comment by AIIX** [comment]
This is the commit info:

Merge: 5d93115 0a6577f
Author: TheAssassin <theassassin@assassinate-you.net>
Date:   Thu Apr 26 20:03:46 2018 +0200

Merge pull request #759 from AppImage/TheAssassin/improve-runtime-debug
    
Improve runtime-debug build

Also:
zlib1g-dev is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).

**Comment by AIIX** [comment]
Did a git checkout on a much older commit ~25days ago: git checkout 85f74cff07249613fe19ac0f96fb6c06306181b2 - this builds fine

**Comment by TheAssassin** [comment]
No, this is the latest commit. I think we have to `git bisect` this issue, though. I'll build the project myself on an ARM system. However, I doubt this is limited to the armhf platform.

**Comment by TheAssassin** [comment]
I just got another idea how to "remote debug" this issue. Can you please re-build with `make VERBOSE=1` and post the full build command?

**Comment by probonopd** [comment]
Thanks @AIIX. Just to be clear, is 85f74cf just a random example for a working commit, or it it the last commit that was still working for those platforms?

**Comment by AIIX** [comment]
@probonopd 85f74cff07249613fe19ac0f96fb6c06306181b2 was a random commit I checked out, had to get some test AppImages going for Arm platform guess got lucky with this commit, so haven't had the chance to check out any commit post that as building this stuff on qemu arm emulation takes significant time, although there seems to be a significant difference in the sizes of the AppImage between x86 (the latest AppImageKit) and armhf/arm64 AppImages (using the older commit), I am not sure what the AppImage produced by a older commit could be missing out on. I am using this in conjunction with the latest linuxdeployqt and Cmake 3.11

**Comment by TheAssassin** [comment]
@AIIX next time, please don't call make with `-j`, and have it run everything sequentially. Makes it a lot easier to read the log.

**Comment by TheAssassin** [comment]
@AIIX I'm still waiting for the verbose log. I can't help you if you don't help me help you.
